gdpr/impl.go

```diff
@@ -251,7 +244,7 @@ func (p *permissionsImpl) parseVendor(ctx context.Context, vendorID uint16, cons
 		return
 	}
 
-	vendorList, err := p.fetchVendorList[version](ctx, parsedConsent.VendorListVersion())
```

Comment:
`fetchVendorList` no longer needs to be a map because the only version we support now is TCF2 given TCF1 was removed a few months ago.